pandas/core/generic.py

```diff
@@ -642,6 +644,142 @@ def _set_axis(self, axis, labels):
         self._data.set_axis(axis, labels)
         self._clear_item_cache()
 
+    def set_index(self, keys, drop=True, append=False, inplace=False,
+                  verify_integrity=False):
+        """
+        Set the index (row labels) using one or more given arrays (or labels).
```

Comment:
i am not sure adding a doc-string actually does anything as you have doc-strings in the subclasses. you are welcome / encouraged to use the appender though (e.g. define the template here) and use it in the subclasses, this pattern is already used heavily, e.g. reindex